### PR TITLE
Fix post model deletes tags when editing post only

### LINKED ISSUE #6919: Editing a post without providing tags results in disaster
If you try to edit a post model without providing the array of tags currently associated with that post, the tags will all be removed.

Essentially, the post model treats an edit operation on a post object with no `tags` property as if you sent `tags: []`. What should be happening, is in the case that the `tags` property is emitted, this should be treated as if we want to do NO operations on tags.

### Steps to Reproduce (code)

The simplest way to reproduce this in the code is to do something like

`models.Posts.edit({title: "My title"}, {id: 1})`

This *should* just edit the title. In fact, it edits the title and deletes the posts.

Meanwhile, elsewhere in the model layer, doing `models.User.edit({name: 'Hannah'}, {id: 1})` would edit my name, but it would also correctly NOT delete all my roles or permissions!!

### Steps to Reproduce (API)

This can also be reproduce in the API, by getting a bearer token from a valid authenticated request with the admin panel.

Then making a PUT request to `/ghost/api/v0.1/posts/x` where the ID is a valid ID of a post. The body must be set to a full post object with a matching ID, but you can exclude providing a tags array, which should result in the post being updated but the tags being left alone.

This also results in the tags being deleted. 

### Technical details:

* Ghost Version: 0.8.8

### PR DESCRIPTION
**Pull request by ErisDS**
fixes #6919
- adds tests at both the model (using withRelated) and API (using includes) layers
- resolves the problem by guarding for `tags` being undefined:
  - If it is undefined, we don't do any tag processing
  - To delete all tags, you would provide `tags: []`

TODO:
- [x] add tests to check correct behaviour of an empty array
- [x] check that behaviour is the same if the `tags` property is set explicitly to undefined, or not provided

### CODE REVIEW

File: core/server/models/post.js [hunk 1]
```diff
@@ -164,18 +165,22 @@ Post = ghostBookshelf.Model.extend({
...
 
-        // keep tags for 'saved' event and deduplicate upper/lowercase tags
-        this.myTags = [];
-
-        _.each(tagsToCheck, function each(item) {
-            for (i = 0; i < self.myTags.length; i = i + 1) {
-                if (self.myTags[i].name.toLocaleLowerCase() === item.name.toLocaleLowerCase()) {
-                    return;
+        // If we have a tags property passed in
+        if (!_.isUndefined(tagsToCheck) && !_.isNull(tagsToCheck)) {
```
Comment (ErisDS) [inline core/server/models/post.js]: Added `&& !_.isNull()` here, to also catch the null case at the model layer. This makes it consistent with the API layer & follows on from the conversation in #dev about only allowing empty arrays to delete all tags being much safer.

**Comment by ErisDS** [comment]
Have extended the tests & added handling for the null case. Consider this ready once the build is green 😁

File: core/test/integration/api/api_posts_spec.js [hunk 2]
```diff
@@ -584,4 +584,205 @@ describe('Post API', function () {
...
+
+            // Step 1, fetch a post from the API with tags
+            PostAPI.read(_.extend({}, options, includeOptions)).then(function (results) {
+                var postWithoutTags = results.posts[0];
+                should.exist(results.posts[0]);
+                should.exist(results.posts[0].tags);
+                results.posts[0].tags.should.have.lengthOf(2);
+
+                // Save the tags for testing against later
+                startTags = _.clone(results.posts[0].tags);
```
Comment (kirrg001) [inline core/test/integration/api/api_posts_spec.js]: unused in this test case

File: core/test/integration/api/api_posts_spec.js [hunk 3]
```diff
@@ -584,4 +584,201 @@ describe('Post API', function () {
...
+                should.exist(results.posts[0]);
+                should.exist(results.posts[0].tags);
+                results.posts[0].tags.should.have.lengthOf(2);
+                results.posts[0].tags.should.eql(startTags);
+
+                done();
+            }).catch(done);
+        });
+
+        it('should update post & not delete tags with `tags` set to null', function (done) {
```
Comment (kirrg001) [inline core/test/integration/api/api_posts_spec.js]: this test looks like a duplicate?

Comment (kirrg001) [inline core/test/integration/api/api_posts_spec.js]: 💆
